### PR TITLE
feat: add spec for copilot CLI

### PR DESCRIPTION
**Pull request by miketheman**
The AWS Copilot CLI is a great CLI tool.
Refs: https://aws.github.io/copilot-cli

This is a first scaffold for the main parts of the subcommands, with
plenty of room to grow to fill out the rest of the command structure.

Followed (and tweaked) the cobra guide, and passed through lint:fix.
Refs: https://fig.io/docs/guides/autocomplete-for-teams/cobra

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature - new spec

**What is the current behavior? (You can also link to an open issue here)**
n/a

**What is the new behavior (if this is a feature change)?**
n/a

**Additional info:**

Another interesting "problem" was - how to express the suggestions, which is an array of strings, to retain the quotation once selected? I tried a couple of variations, but adding `'"thing"'` emitted both the single and double quotes to the command line, and that's not what I was expecting.

### CODE REVIEW

File: src/copilot.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          args: {
+            name: "string",
+          },
+        },
+        {
+          name: "--deploy",
+          description: "Deploy service/job to a 'test' environment",
+        },
+        {
+          // TODO: Is there a way to make this mutually exclusive with `-i`?
```
Comment (fedeci) [inline src/copilot.ts:31]: `exclusiveOn: ` seems to be what you are looking for, you have to add it to both -i and -d

Comment (miketheman) [reply src/copilot.ts:31]: Thanks! No clue why I didn't see that in the docs last night.

Comment (miketheman) [reply src/copilot.ts]: @fedeci See next commit - I followed [the docs](https://fig.io/docs/reference/option#exclusiveon).

Works well with the `--<option>` , but doesn't appear to work as well with the shorter `-I`/`-d` - is that something in the spec syntax, or something in the core?

File: src/copilot.ts [hunk 2]
```diff
@@ -28,19 +28,20 @@ const completionSpec: Fig.Spec = {
           description: "Deploy service/job to a 'test' environment",
         },
         {
-          // TODO: Is there a way to make this mutually exclusive with `-i`?
           name: ["-d", "--dockerfile"],
           description: "Path to the Dockerfile",
           args: {
             name: "string",
           },
+          exclusiveOn: ["-i", "--image"],
```
Comment (fedeci) [inline src/copilot.ts:36]: I'm pretty sure `-i` is enough
@QuiiBz am I right?
```suggestion
          exclusiveOn: ["-i", "--image"],
```

**Comment by miketheman** [comment]
Hey gang! I think this is ready for review - it's a bit bigger than before, and includes a lot more functionality.

**Review by mschrage — APPROVED** [review]
Thanks so much for this contribution @miketheman! It looks great :)

**Comment by miketheman** [comment]
Thanks @mschrage ! I updated the base branch, so it's all green now!

**Comment by fedeci** [comment]
Great, thanks!

**Pull request merged**